src/components/JdsDataTable/mixins/select.js

```diff
@@ -0,0 +1,144 @@
+const select = {
+
+  data() {
+    return {
+      selectedItems: [],
+      selectAll: false
+    }
+  },
+  methods: {
+
+    /**
+     * Check if an Object has a property
+     * @param {Object} item 
+     * @param {String, Number} property 
+     * @returns {Boolean}
+     */
+    hasProperty(item, property) {
+      return Object.prototype.hasOwnProperty.call(item, property)
+    },
+
+    /**
+     * Triggered when checkbox on table
+     * header is clicked
+     */
+    onSelectAll() {
+
+      if (!this.isKeyExist) {
+        return
+      }
+
+      if (this.isSelectingAllItemsWithinPage) {
+        this.items.forEach((item) => {
+          this.onRowSelectChange(item, true)
+        })
+
+        this.emitChanges()
+      } else {
+        this.items.forEach((item) => {
+          this.onRowSelectChange(item)
+        })
+
+        this.emitChanges()
+      }
+    },
+
+    /**
+     * Triggered when checkbox on the table body
+     * is clicked
+     * @param {Object} item 
+     * @param {Boolean} bool
+     */
+    onRowSelectChange(item, bool) {
+
+      if (!this.isKeyExist) {
+        return
+      }
+
+      /**
+       * Toggling insert and remove item
+       */
+      if (typeof bool === 'boolean' && bool === true) {
+        if (this.isSelected(item)) {
+          const array = this.selectedItems.filter(value => value !== item[this.itemKey])
+          this.selectedItems = [...array]
+        } else {
+          this.selectedItems.push(item[this.itemKey])
+        }
+        return
+      }
+
+      /**
+       * Force push unselected item
+       */
+      if (!this.isSelected(item)) {
+        this.selectedItems.push(item[this.itemKey])
+      }
+    },
+
+    /**
+     * Emmit an event with 
+     * array of selected items
+     */
+    emitChanges() {
+      if (!this.isKeyExist) {
+        return
+      }
+      this.$emit('change:select', this.selectedItems)
+    },
+
+    /**
+     * Check whether the item provided is currently
+     * being selected
+     * @param {Object} item 
+     * @returns {Boolean}
+     */
+    isSelected(item) {
+      return this.selectedItems.includes(item[this.itemKey])
+    },
+
+  },
+  computed: {
+
+    /**
+     * Check whether the itemKey provided are exists
+     * within the items 
+     * @returns {Boolean}
+     */
+    isKeyExist() {
+      return this.items.every(item => this.hasProperty(item, this.itemKey))
+    },
```

Comment:
say, we have an array with 1000 elements, in which one of them doesn't have an `itemKey` property, that would break the selectable/checkbox/toggle-able functionality, IIRC?.

if so, `itemKey` prop documentation should probably warn/mandate the developer to fulfill the requirement above.

Reply:
yes, it's not throwing an error but the component will not emit events. 
the default value of `itemKey` props is `id`, but on some cases it might be named differently. 

yes, I agree there should be more explanation and better documentation about this, thank you @adrianpdm

Reply:
hello, this issues already fixed [here](https://github.com/jabardigitalservice/jds-design-system-vue-lib/pull/90/commits/5430b42aa9a600ef3a166cef983aabba4c93a304) @adrianpdm